FreeRTOS_IP.c

```diff
@@ -4184,7 +4267,7 @@ const char * FreeRTOS_strerror_r( BaseType_t xErrnum,
 
         default:
             /* Using function "snprintf". */
-            ( void ) snprintf( pcBuffer, uxLength, "Errno %d", ( printf_signed ) xErrnum );
+            ( void ) snprintf( pcBuffer, uxLength, "Errno %d", xErrnum );
```

Comment:
We've removed snprintf from other code due to MISRA 21.6.  We should consider doing so here too.  The justification provided in MISRA.md is not persuasive.

Reply:
Luckily there are very few occurrences of `snprintf()`. What alternative would you recommend? I suppose that `itoa()` is worse than `snprintf()`?

Reply:
Is this list not already exhaustive?  What value could xErrnum have that is not one of the cases?

Reply:
Yes good answer, in this case. All other uses of `snprintf()` ( not many ) have to do with debug logging, which is not an essential part of the library.

Reply:
There was a second snprintf() with a `%s`: I have replaced that with a `strncpy()`:
~~~
    if( uxLength > 0U )
    {
        ( void ) strncpy( pcBuffer, pcName, uxLength - 1U );

        /* Make sure the result is nul-terminated. */
        pcBuffer[ uxLength - 1U ] = '\0';
    }
~~~